src/Evolution/Executables/CurvedScalarWave/EvolveCurvedScalarWave.hpp

```diff
@@ -0,0 +1,349 @@
+          tmpl::at<typename factory_creation::factory_classes, Event>>>>;
+
+  static constexpr bool use_filtering = true;
+
+  using step_actions = tmpl::flatten<tmpl::list<
+      evolution::dg::Actions::ComputeTimeDerivative<EvolutionMetavars>,
+      tmpl::conditional_t<
+          local_time_stepping,
+          tmpl::list<evolution::Actions::RunEventsAndDenseTriggers<>,
+                     evolution::dg::Actions::ApplyBoundaryCorrections<
+                         EvolutionMetavars>>,
+          tmpl::list<evolution::dg::Actions::ApplyBoundaryCorrections<
+                         EvolutionMetavars>,
+                     Actions::RecordTimeStepperData<>,
+                     evolution::Actions::RunEventsAndDenseTriggers<>,
+                     Actions::UpdateU<>>>,
+      tmpl::conditional_t<
+          use_filtering,
+          dg::Actions::Filter<Filters::Exponential<0>,
+                              tmpl::list<CurvedScalarWave::Tags::Psi,
+                                         CurvedScalarWave::Tags::Pi,
+                                         CurvedScalarWave::Tags::Phi<Dim>>>,
+          tmpl::list<>>>>;
+
+  enum class Phase {
+    Initialization,
+    RegisterWithObserver,
+    Register,
+    InitializeTimeStepperHistory,
+    LoadBalancing,
+    WriteCheckpoint,
+    Evolve,
+    Exit
+  };
+
+  static std::string phase_name(Phase phase) {
+    if (phase == Phase::LoadBalancing) {
+      return "LoadBalancing";
+    }
+    ERROR(
+        "Passed phase that should not be used in input file. Integer "
+        "corresponding to phase is: "
+        << static_cast<int>(phase));
+  }
+
+  using phase_changes =
+      tmpl::list<PhaseControl::Registrars::VisitAndReturn<EvolutionMetavars,
+                                                          Phase::LoadBalancing>,
+                 PhaseControl::Registrars::CheckpointAndExitAfterWallclock<
+                     EvolutionMetavars>>;
+
+  using initialize_phase_change_decision_data =
+      PhaseControl::InitializePhaseChangeDecisionData<phase_changes>;
+
+  using phase_change_tags_and_combines_list =
+      PhaseControl::get_phase_change_tags<phase_changes>;
+
+  using const_global_cache_tags =
+      tmpl::list<initial_data_tag, Tags::EventsAndTriggers,
+                 PhaseControl::Tags::PhaseChangeAndTriggers<phase_changes>>;
+
+  using dg_registration_list =
+      tmpl::list<observers::Actions::RegisterEventsWithObservers>;
+
+  using initialization_actions = tmpl::list<
+      Actions::SetupDataBox,
+      Initialization::Actions::TimeAndTimeStep<EvolutionMetavars>,
+      evolution::dg::Initialization::Domain<volume_dim>,
+      Initialization::Actions::NonconservativeSystem<system>,
+      evolution::Initialization::Actions::SetVariables<
+          domain::Tags::Coordinates<Dim, Frame::ElementLogical>>,
+      Initialization::Actions::TimeStepperHistory<EvolutionMetavars>,
+      CurvedScalarWave::Actions::InitializeConstraintDampingGammas<volume_dim>,
```

Comment:
This is where you'd use the `AddSimpleTags` and/or `AddComputeTags` actions with the mutators, just like on line 253 below. Hopefully in the long run we can move most or all initialize actions to mutators just to simplify testing them and reduce the large amount of boilerplate code that actions need...